spikeextractors/extractors/shybridextractors/shybridextractors.py

```diff
@@ -0,0 +1,174 @@
+import os
+
+import numpy as np
+
+from spikeextractors import RecordingExtractor
+from spikeextractors import SortingExtractor
+from spikeextractors.extractors.bindatrecordingextractor import BinDatRecordingExtractor
+from spikeextractors.extraction_tools import save_to_probe_file
+
+try:
+    import hybridizer.io as sbio
+    import hybridizer.probes as sbprb
+    HAVE_SBEX = True
+except ImportError:
+    HAVE_SBEX = False
+
+class SHYBRIDRecordingExtractor(RecordingExtractor):
+    extractor_name = 'SHYBRIDRecordingExtractor'
+    installed = HAVE_SBEX
+    extractor_gui_params = [
+        {'name': 'recording_fn', 'type': 'file', 'title': "Full path to hybrid recording (.bin)"},
+    ]
+
+    def __init__(self, recording_fn):
+        RecordingExtractor.__init__(self)
+
+        # load params file related to the given shybrid recording
+        params = sbio.get_params(recording_fn)['data']
+
+        # create a shybrid probe object
+        probe = sbprb.Probe(params['probe'])
+        nb_channels = probe.total_nb_channels
+
+        # translate the byte ordering
+        # TODO still ambiguous, shybrid should assume time_axis=1, since spike interface makes an assumption on the byte ordering
+        byte_order = params['order']
+        if byte_order == 'C':
+            time_axis = 1
+        elif byte_order == 'F':
+            time_axis = 0
+
+        # piggyback on binary data recording extractor
+        self._bindatext = BinDatRecordingExtractor(recording_fn,
+                                                   params['fs'],
+                                                   nb_channels,
+                                                   params['dtype'],
+                                                   time_axis=time_axis)
+
+    def get_channel_ids(self):
+        return self._bindatext.get_channel_ids()
+
+    def get_num_frames(self):
+        return self._bindatext.get_num_frames()
+
+    def get_sampling_frequency(self):
+        return self._bindatext.get_sampling_frequency()
+
+    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
+        return self._bindatext.get_traces(channel_ids=channel_ids,
+                                          start_frame=start_frame,
+                                          end_frame=end_frame)
+
+    @staticmethod
+    def write_recording(recording, save_path, initial_sorting_fn, dtype='float32'):
+        """ Convert and save the recording extractor to SHYBRID format
+
+        parameters
+        ----------
+        recording: RecordingExtractor
+            The recording extractor to be converted and saved
+        save_path: str
+            Full path to desired target folder
+        initial_sorting_fn: str
+            Full path to the initial sorting csv file (can also be generated
+            using write_sorting static method from the SHYBRIDSortingExtractor)
+        dtype: dtype
```

Comment:
you can also infer the dtype from the recording. We'll make a method of the base extractor soon (so this is ok)

Reply:
I don't fully get this comment. Do you want me to take any action regarding this topic right now?

Reply:
you could automatically extract the dtype as follows:
`dtype = recording.get_traces(start_frame=0, end_frame=1).dtype`

I think it would be better. I agree that my previous comment was very confusing!

Reply:
Here the dtype reflects the type that user wants to save the data to, this could be potentially different from the data type that is used internally in spikeinterface.

Reply:
ah! Ok then leave it there :)